include/mbgl/style/layer.hpp

```diff
@@ -65,6 +66,14 @@ struct LayerTypeInfo {
     const enum class TileKind : uint8_t { Geometry, Raster, RasterDEM, NotRequired } tileKind;
 };
 
+struct LayerProperty {
```

Comment:
What about s/LayerProperty/StyleProperty/ ? This could be used for light properties too.